arch/risc-v/src/esp32c3/esp32c3_rtc_lowerhalf.c

```diff
@@ -0,0 +1,575 @@
+
+  /* Remember the callback information */
+
+  cbinfo            = &priv->cbinfo[alarminfo->id];
+  cbinfo->cb        = alarminfo->cb;
+  cbinfo->priv      = alarminfo->priv;
+
+  /* Set the alarm */
+
+  lowerinfo.as_id   = alarminfo->id;
+  lowerinfo.as_cb   = rtc_alarm_callback;
+  lowerinfo.as_arg  = priv;
+
+  /* Convert the RTC time to a timespec (1 second accuracy) */
+
+  lowerinfo.as_time.tv_sec  = mktime((FAR struct tm *)&alarminfo->time);
+  lowerinfo.as_time.tv_nsec = 0;
+
+  /* And set the alarm */
+
+  ret = up_rtc_setalarm(&lowerinfo);
+  if (ret < 0)
+    {
+      cbinfo->cb   = NULL;
+      cbinfo->priv = NULL;
+    }
+
+  return ret;
+}
+#endif /* CONFIG_RTC_ALARM */
+
+/****************************************************************************
+ * Name: rtc_setrelative
+ *
+ * Description:
+ *   Set a new alarm relative to the current time. This function implements
+ *   the setrelative() method of the RTC driver interface
+ *
+ * Input Parameters:
+ *   lower - A reference to RTC lower half driver state structure
+ *   alarminfo - Provided information needed to set the alarm
+ *
+ * Returned Value:
+ *   Zero (OK) is returned on success; a negated errno value is returned
+ *   on any failure.
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_RTC_ALARM
+static int rtc_setrelative(FAR struct rtc_lowerhalf_s *lower,
+                           FAR const struct lower_setrelative_s *alarminfo)
+{
+  struct lower_setalarm_s setalarm;
+  time_t seconds;
+  int ret = -EINVAL;
+  irqstate_t flags;
+
+  DEBUGASSERT(lower != NULL && alarminfo != NULL);
+  DEBUGASSERT((RTC_ALARM0 <= alarminfo->id) &&
+              (alarminfo->id < RTC_ALARM_LAST));
+
+  if (alarminfo->reltime > 0)
+    {
+      flags = spin_lock_irqsave(NULL);
+
+      seconds = alarminfo->reltime;
+      gmtime_r(&seconds, (FAR struct tm *)&setalarm.time);
+
+      /* The set the alarm using this absolute time */
+
+      setalarm.id   = alarminfo->id;
+      setalarm.cb   = alarminfo->cb;
+      setalarm.priv = alarminfo->priv;
+      ret = rtc_setalarm(lower, &setalarm);
+
+      spin_unlock_irqrestore(NULL, flags);
+    }
+
+  return ret;
+}
+#endif /* CONFIG_RTC_ALARM */
+
+/****************************************************************************
+ * Name: rtc_cancelalarm
+ *
+ * Description:
+ *   Cancel the current alarm.  This function implements the cancelalarm()
+ *   method of the RTC driver interface
+ *
+ * Input Parameters:
+ *   lower - A reference to RTC lower half driver state structure
+ *   alarmid - the alarm id
+ *
+ * Returned Value:
+ *   Zero (OK) is returned on success; a negated errno value is returned
+ *   on any failure.
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_RTC_ALARM
+static int rtc_cancelalarm(FAR struct rtc_lowerhalf_s *lower,
+                                                int alarmid)
```

Comment:
Place it in the line before or fix the alignment

Reply:
Done